codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSHTTPProtocolCustomizations.kt

```diff
@@ -58,7 +60,13 @@ abstract class AWSHTTPProtocolCustomizations : DefaultHTTPProtocolCustomizations
     }
 
     override fun renderInternals(ctx: ProtocolGenerator.GenerationContext) {
-        AuthSchemeResolverGenerator().render(ctx)
+        AuthSchemeResolverGenerator { authOptionName, writer ->
+            writer.write(
+                "$authOptionName.identityProperties.set(key: \$N.internalSTSClientKey, value: \$N())",
+                AWSSDKIdentityTypes.InternalClientKeys,
+                InternalClientTypes.IdentityProvidingSTSClient,
+            )
+        }.render(ctx)
```

Comment:
This is where we use the new extension point added in companion smithy-swift PR. Allows us to construct & add `IdentityProvidingSTSClient` struct to `identityProperties`, without leaking type info in generic smithy codegen.